app/javascript/controllers/pair_requests_form_controller.js

```diff
@@ -0,0 +1,16 @@
+import { Controller } from "@hotwired/stimulus"
+import { get } from "@rails/request.js"
+// Connects to data-controller="pair-requests-form"
+export default class extends Controller {
```

Comment:
it may not be possible for this, but the hope for stimulus controllers is that they are both focused/small, as well as general enough to be used in multiple places in the app.

I wonder whether it might make sense to refactor this to be more general? Naming it might be hard (I initially thought of `selectOptionTurboStreamTrigger`, but I don't think that's a particularly great name).

One way I might think of making it more general is if the url and param_name are passed to the controller as a `value` attribute, which means you don't need to hardcode it within the controller itself.

This could simplify the change code to be something like:
```js
  change(event) {
    let selectOption = event.target.selectedOptions[0].value

    get(`${urlValue}?${paramValue}=${selectOption},{
      responseKind: "turbo-stream"
    })
  }
```

What do you think?

Reply:
Thank you for the suggestion @afogel, I would definitely want to refactor my approach to reflect that ideal for stimulus controllers to be focused/reusable and your example will really help! I'm trying out some other approaches rn and once I get my ducks in an order I'll see if I can refactor along those lines.